### DISCUSSION TITLE
Instability issue

### DISCUSSION THREAD

**Discussion by staticgroup** [OpenSurv/OpenSurv#10] [General]
during my testing I have noticed the major issue that if I make a change to a stream ie: restart a camera, change rtsp setting on a device it crashed opensurv instantly like I mean i click it crashed. its consistent across nest mtx and android based ip cameras and ubuiquti unvr.

**Comment by SvenVD** [comment]
Is there something in the logs? Can you describe the crash, what is shown on the screen? Can you enable [debug logging](https://github.com/OpenSurv/OpenSurv/blob/4a8f282b2aa747dd5c5999a754336a29884563c6/README.md?plain=1#L117) and trigger the crash again?
Thanks

**Comment by staticgroup** [comment]
Nothing in logs for service or standard check. I'll turn on debugging and trigger again. Screen going to blinking cursor on mon1 and mon2 is fully blank.  Suspect service terms

**Comment by SvenVD** [comment]
I managed to reproduce it, will provide a fix later on

**Comment by staticgroup** [comment]
ok great any temp suggestions?

**Comment by staticgroup** [comment]
Did some logging . Hopefully it can help as I now have some streams not responding and need a service restart 
[main6.log](https://github.com/user-attachments/files/17774996/main6.log)
[main7.log](https://github.com/user-attachments/files/17774997/main7.log)
[main8.log](https://github.com/user-attachments/files/17774998/main8.log)
[main1.log](https://github.com/user-attachments/files/17774999/main1.log)
[main2.log](https://github.com/user-attachments/files/17775001/main2.log)
[main3.log](https://github.com/user-attachments/files/17775002/main3.log)
[main4.log](https://github.com/user-attachments/files/17775004/main4.log)
[main5.log](https://github.com/user-attachments/files/17775005/main5.log)
Note: had to rename them as the logning function adds a .# instead of main#.log

**Comment by SvenVD** [comment]
Can you test the latest version:
https://github.com/OpenSurv/OpenSurv/releases/tag/1.2beta1
It should be fixed. On remote stream setting change, the related mpv video should disappear and after some seconds it should reappear with new settings.

**Comment by staticgroup** [comment]
Have ran today will let you know updates. thank you!

**Comment by staticgroup** [comment]
Straight off the bat along of reconnecting for 5-10seconds at time accross my ubiquti cameras. The beelink is doing well poerformance wise only 28% cpu and 30% memory and recently after staying crashed for about 30min it went to a lock screen which is very odd considering its ubuntu server so should be a console screen not a gui based password screen. 
[logs rtsp.zip](https://github.com/user-attachments/files/17788525/logs.rtsp.zip)

**Reply by SvenVD** [reply]
> Straight off the bat along of reconnecting for 5-10seconds at time accross my ubiquti cameras. The beelink is doing well poerformance wise only 28% cpu and 30% memory and recently after staying crashed for about 30min it went to a lock screen which is very odd considering its ubuntu server so should be a console screen not a gui based password screen. [logs rtsp.zip](https://github.com/user-attachments/files/17788525/logs.rtsp.zip)

The installer installs xfce window manager which OpenSurv needs and the gui comes along with it. It is also helpful for less experienced linux users, they can press "q" and login with a GUI.

**Comment by SvenVD** [comment]
Where there any actions by you around the time of crash? Can you share your config? ( remove any passwords)

**Reply by staticgroup** [reply]
no actions by me at the time.

**Comment by SvenVD** [comment]
Ok I think I see now in the logs, this is a different problem then the bug with remote settings change you described earlier.  This is specifically related only when you have a second monitor attached. I will try to reproduce the issue.

**Comment by SvenVD** [comment]
I uploaded 1.2beta2 which should fix your last issue with second monitor. Can you test it? Thanks

**Reply by staticgroup** [reply]
yes will do sorry daytime here was playing with the kids

**Reply by staticgroup** [reply]
definately see a increase in stability straight away loads smoother too for the second screen (screen 1 was smoother with 1.1)

**Reply by staticgroup** [reply]
scratch that, reoccuring alot later , about 4min into running the service but it recovers 2-3x faster
will grab logs and config files

**Reply by staticgroup** [reply]
[logs rtsp.zip](https://github.com/user-attachments/files/17788631/logs.rtsp.zip)
Logs and Config attached I slightly redacted it,

**Reply by staticgroup** [reply]
even though retrospectively that info is in the logs haha

**Reply by SvenVD** [reply]
> scratch that, reoccuring alot later , about 4min into running the service but it recovers 2-3x faster will grab logs and config files

Can you describe what is reoccuring and a rough timestamp would also be handy, according to the new debug logs at least the complete crash to login screen seems to be fixed now. Can you confirm? Thanks for the extensive testing!

**Reply by staticgroup** [reply]
timestamp is hard as its continuos . happy to test as much as possible. I have now setup a second beelink so am a avid user now :D

**Comment by SvenVD** [comment]
Yet another version https://github.com/OpenSurv/OpenSurv/releases/tag/1.2beta3 with more extensive logging to know what is causing your last problem.

**Comment by staticgroup** [comment]
noticed there was a beta 4 so jumped on that as the latest release. 1. Will provide logs on that 2. happy to revert to beta3 at your command as required.

**Comment by staticgroup** [comment]
crashing on monitor2 seems more agressive on beta4 (could be my imagination) will run for another 45min before posting logs.

**Comment by staticgroup** [comment]
[logs.zip](https://github.com/user-attachments/files/17794045/logs.zip)
logs attached

**Comment by SvenVD** [comment]
My guess is that these three streams are slower to connect to, could you increase `probe_timeout` and `timeout_waiting_for_init_stream` timeouts thresholds for those streams:

- rtsp://192.168.1.209:8554/kitchen
- rtsp://192.168.1.209:8554/livingroom
- rtsp://192.168.40.38:8080/h264_pcm.sdp

To something like

- `timeout_waiting_for_init_stream: 30` [example](https://github.com/OpenSurv/OpenSurv/blob/4a8f282b2aa747dd5c5999a754336a29884563c6/surveillance/etc/monitor1.yml#L107)
- `probe_timeout: 30` [example](https://github.com/OpenSurv/OpenSurv/blob/4a8f282b2aa747dd5c5999a754336a29884563c6/surveillance/etc/monitor1.yml#L91)

If the streams are really dead then such high timeouts slows everything down, but this is only for sake of testing and finding the root cause.

Additionally if that does not work,
Try removing those three streams and see what happens

Or try connecting to them with mpv directly and see what happens

Thanks

**Comment by staticgroup** [comment]
removing all the non unifi streams stablized it. So need to tweak something. I am now testing my android streams at lowest res. Seems to be stablish. Sorry busy few days

**Comment by SvenVD** [comment]
If you tweak the non unifi with the timeouts settings I think you have a high chance of stabilizing it

**Comment by SvenVD** [comment]
Did you have any chance of testing with the higher timeout settings for those particular streams?
